File: JavaScript/5kyu/GapInPrimes.js
function gap(g, m, n) {
  var lastPrime = 0;
  for (let i = m; i <= n; i++) {
    if (isPrime(i)) {
      if (i - lastPrime == g) return [lastPrime, i];

      lastPrime = i;
    }
  }
  return null;
}

const isPrime = (n) => {
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
};

var result = gap(8, 300, 400);
console.log(result);

export { gap };

File: Python/5kyu/GapInPrimes.py
import math


def gap(g, m, n):
    last_prime = 0

    for p in range(m, n+1):
        if is_prime(p):
            if p - last_prime == g:
                return [last_prime, p]

            last_prime = p

    return []


def is_prime(n):
    for x in range(2, int(math.sqrt(n))):
        if n % x == 0:
            return False

    return True


print(gap(3, 3, 10))
